### PR TITLE
Tls+auth

### PR DESCRIPTION
**Pull request by jmagoon**
- tls auth code 
- aws support
- remote logging

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -345,14 +350,30 @@ func main() {
...
 		router.Use(func(ctx *gin.Context) {
-			if ctx.Request.Method != "OPTIONS" {
+			if ctx.Request.Method == "OPTIONS" {
+				ctx.Next()
+				return
+			}
+
+			// Check if it's a WebSocket upgrade request
+			if isWebSocketRequest(ctx.Request) {
+				token := ctx.Query("token")
```
Comment (plparent) [inline main.go:360]: **REALLY BAD** : This creates a token leak.

File: token.go [hunk 2]
```diff
@@ -67,6 +67,50 @@ func validateToken(ctx *gin.Context, folder string) (int, string) {
...
+	if !token.Valid {
+		log.Printf("Invalid token")
+		return http.StatusUnauthorized, ""
+	}
+
 	sub, _ := token.Claims.GetSubject()
 	return http.StatusOK, sub
 }
+
+func validateSocketToken(token string, folder string) (int, string) {
```
Comment (plparent) [inline token.go:80]: Duplicate function: I would recommend refactoring both into one.

File: credentials.go [hunk 3]
```diff
@@ -69,31 +69,37 @@ func (endpoint *CloudEndpoint) ResolveEndpoint(ctx context.Context, params s3.En
...
 
 	if cfg.IsLocal {
 		awsAccessKey = "AKIAIOSFODNN7EXAMPLE"
 		awsSecretKey = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
-		endpoint = &LocalEndpoint{Bucket: cfg.BucketName, S3Port: cfg.Local.BucketPort}
+		endpointResolver = &LocalEndpoint{Bucket: cfg.BucketName, S3Port: cfg.Local.BucketPort}
 	} else {
 		awsAccessKey = cfg.Cloud.AWS.AccessKey
 		awsSecretKey = cfg.Cloud.AWS.SecretKey
-		endpoint = &CloudEndpoint{Address: "s3-us-east-2.amazonaws.com", Bucket: cfg.BucketName}
```
Comment (plparent) [inline credentials.go]: I'm a bit confused, but where do you set the bucket?

**Review by plparent — CHANGES_REQUESTED** [review]

File: main.go [hunk 4]
```diff
@@ -594,7 +602,12 @@ func main() {
 		var response []ResponsePipelineExecution
 		for _, execution := range res {
 			logOutput := []string{}
-			tempLog := filepath.Join(os.TempDir(), execution.Executionid+".log")
+			logDir, exists := os.LookupEnv("ZETAFORGE_LOGS")
+			if !exists {
+				logDir = os.TempDir()
+			}
+			tempLog := filepath.Join(logDir, execution.Executionid+".log")
+
```
Comment (plparent) [inline main.go:649]: Would it make more sense to set logDir at the start of main? Also I feel the check should be local vs cloud, but maybe I'm missing something.

**Review by plparent — CHANGES_REQUESTED** [review]

File: credentials.go (hunk 3 above)
Comment (jmagoon) [reply credentials.go]: The bucket is a required parameter in the get / put / list requests:

params := &s3.PutObjectInput{
		Bucket: aws.String(cfg.BucketName),
		Key:    aws.String(key),
		Body:   file,
	}
	
The s3Client has a default resolver, we only need to set the resolver to something custom when we have our own endpoint

File: token.go (hunk 2 above)
Comment (jmagoon) [reply token.go]: done

File: main.go (hunk 4 above)
Comment (jmagoon) [reply main.go:643]: Not for this type of log, which is not the system log, but the log for the execution. I agree with you though, in the future we would want to do log handling but split our server logs from execution logs

File: main.go (hunk 1 above)
Comment (jmagoon) [reply main.go]: swapped to protocol header

File: credentials.go (hunk 3 above)
Comment (plparent) [reply credentials.go]: So do we assume that all cloud endpoints are always aws s3?

Comment (jmagoon) [reply credentials.go]: Good point, I'll put back the endpoint line. Using the default resolver was testing if it was incorrect due to the 301 upload error

**Review by plparent — APPROVED** [review]

**Pull request merged**